misc/swagger-ballerina/modules/swagger-to-ballerina-generator/src/test/java/CodeGeneratorTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.ballerinalang.swagger.CodeGenerator;
+import org.ballerinalang.swagger.utils.GeneratorConstants.GenType;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Unit tests for {@link org.ballerinalang.swagger.CodeGenerator}
+ */
+public class CodeGeneratorTest {
+    private String testResourceRoot;
+
+    @BeforeClass()
+    public void setup() {
+        testResourceRoot = CodeGeneratorTest.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+    }
+
+    @Test(description = "Test Ballerina skeleton generation when destination path is provided")
+    public void generateSkeletonWithDestination() {
+        String outPath = testResourceRoot + File.separator + "service.bal";
+        String definitionPath = testResourceRoot + File.separator + "petstore.yaml";
+        CodeGenerator generator = new CodeGenerator();
+        try {
+            generator.generate(GenType.SKELETON, definitionPath, outPath);
+            File genFile = new File(outPath);
+            Assert.assertTrue(genFile.exists());
```

Comment:
Shall we check if the generated file content is correct ?

Reply:
Do we need to validate resulting file content in a unit test?